### DISCUSSION TITLE
Add option to pass list of repos as JSON

### DISCUSSION THREAD

**Discussion by okainov** [renovatebot/renovate#12335] [Suggest an Idea]
### What would you like Renovate to be able to do?

Apparently there is `writediscoveredrepos` option available (https://docs.renovatebot.com/self-hosted-configuration/#writediscoveredrepos) which writes list of repos to JSON.

I'd like renovate to be able to read this JSONas input for regular commands (like `renovate [options] --repositories /path/to/file.json`) instead of passing repositories themselves as command-line argument.

Problem background: autodiscover is a) slow and b) doesn't respect Github API limits thus failing on big amount of repos to discover.

### If you have any ideas on how this should be implemented, please tell us here.

-

### Is this a feature you are interested in implementing yourself?

No

**Comment by viceice** [comment]
You can already pass them via env or `config.js`. so please extend your requirements.

**Comment by okainov** [comment]
>You can already pass them via env or config.js. so please extend your requirements.

Passing via env, I believe, doesn't work when amount of repos is quite big (think of 200 repos for example).

I don't use `config.js` on renovate side and don't really want to introduce it. I want to make use of already existing functionality in Renovate producing JSON file.

**Comment by viceice** [comment]
You can also simply dump the `repositories` string array also as `config.json`.

**Comment by viceice** [comment]
So i don't think we need a new option.

**Comment by okainov** [comment]
@viceice okay, but then what's the point of `writediscoveredrepos`? If it produces something which renovate cannot read, then it's quite meaningless. What would be the use case for it if not "prepare repos once for further reuse"?

**Comment by viceice** [comment]
That was a requirement by a single user in #11806 and not a normal renovate option.

It will be used for postprocessing that list and split to pass to mutltiple workers.

**Comment by okainov** [comment]
@viceice well, that's exactly what I'm doing on my side - splitting the list to chunks and passing this to renovate during several runs. And I do think possibility to accept repos in KISS format is something which will be useful.

**Comment by viceice** [comment]
i don't see any missing feature here.

if you process the export anyways, you can create multiple json files with only the processed `repositories` string array and use  `RENOVATE_CONFIG_FILE`  to point there, so renovate reads the repositories list from that file?

**Reply by rarkins** [reply]
I agree about putting it into a `config.json` and have moved this to a Discussion as no feature is required here

**Reply by okainov** [reply]
I can. I can do many workarounds for the tool for sure, that's software which can do everything. But now you'll have a person who does use `config.json` already with proper settings and he'll be forced to now have some copies of this config, some parsing logic updating only `repositories`.

**Reply by rarkins** [reply]
This is why we recommend config.js as it has the flexibility to optionally require json files
